File: tests/fixtures/security/jackson_default_typing/positive.java
package test;

import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.databind.jsontype.impl.LaissezFaireSubTypeValidator;

class MapperFactory {
    ObjectMapper unsafe() {
        new ObjectMapper().enableDefaultTyping();
        return new ObjectMapper().activateDefaultTyping(LaissezFaireSubTypeValidator.instance);
    }
}

File: tests/fixtures/security/jackson_default_typing/positive.kt
package test

import com.fasterxml.jackson.databind.ObjectMapper
import com.fasterxml.jackson.databind.jsontype.impl.LaissezFaireSubTypeValidator

class MapperFactory {
    fun unsafe(): ObjectMapper {
        ObjectMapper().enableDefaultTyping()
        return ObjectMapper().activateDefaultTyping(LaissezFaireSubTypeValidator.instance)
    }
}
